src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -1440,6 +1440,60 @@ void ExtendedDefaultPolicy::NoteDouble(InlineObservation obs, double value)
     m_ProfileFrequency = value;
 }
 
+//------------------------------------------------------------------------
+// BudgetCheck: see if this inline would exceed the current budget
+//
+// Returns:
+//   True if inline would exceed the budget.
+//
+bool ExtendedDefaultPolicy::BudgetCheck() const
+{
+    if (m_IsPrejitRoot)
+    {
+        // Only relevant if we're actually inlining.
+        return false;
+    }
+
+    // The strategy tracks the amout of inlining done so far, so it performs the actual check.
+    const bool overBudget = m_RootCompiler->m_inlineStrategy->BudgetCheck(m_CodeSize);
+
+    if (overBudget)
```

Comment:
You might want to flip this predicate and return early. Getting down to the end of the method you have 3 `return` statements all at different indentations and it makes understanding this flow harder than it needs to be, just return as soon as you know the answer.